server/src/main/java/org/elasticsearch/ingest/IngestService.java

```diff
@@ -86,17 +92,38 @@
 
     public IngestService(ClusterService clusterService, ThreadPool threadPool,
                          Environment env, ScriptService scriptService, AnalysisRegistry analysisRegistry,
-                         List<IngestPlugin> ingestPlugins) {
+                         List<IngestPlugin> ingestPlugins, IndicesService indicesService) {
         this.clusterService = clusterService;
         this.scriptService = scriptService;
+        final IndexNameExpressionResolver resolver = new IndexNameExpressionResolver();
         this.processorFactories = processorFactories(
             ingestPlugins,
             new Processor.Parameters(
                 env, scriptService, analysisRegistry,
                 threadPool.getThreadContext(), threadPool::relativeTimeInMillis,
                 (delay, command) -> threadPool.schedule(
                     command, TimeValue.timeValueMillis(delay), ThreadPool.Names.GENERIC
-                ), this
+                ), this, indexExpression -> {
+                    ClusterState state = clusterService.state();
+                    Index[] resolvedIndices = resolver.concreteIndices(state, IndicesOptions.STRICT_EXPAND_OPEN, indexExpression);
+                    if (resolvedIndices.length != 1) {
+                        throw new IllegalStateException("expression [" + indexExpression + "] can only point to a single concrete index");
+                    }
```

Comment:
should we prevent filtered aliases too?

Reply:
Yes, that makes sense. There is no guarantee that they will be applied in an ingest processor. I will fail with an error if an alias has a filter.